ideascube/serveradmin/wifi.py

```diff
@@ -0,0 +1,183 @@
+from itertools import groupby
+from operator import attrgetter
+from uuid import uuid4
+
+from django.utils.translation import ugettext as _
+
+try:
+    from NetworkManager import (
+        NetworkManager, Settings as NMSettings, NM_DEVICE_TYPE_WIFI)
+
+except Exception as e:
+    import sys
+    sys.stderr.write("Could not import the NetworkManager module: %s\n" % e)
+    sys.stderr.write("Wifi management will not work properly\n")
+    sys.stderr.flush()
+
+    # Set these to None, people will get AttributeError if they try calling
+    # the wifi views, but they have been warned above
+    NetworkManager = None
+    NMSettings = None
+
+    # Hardcode this one, it is a constant anyway in the NetworkManager module,
+    # and hardcoding it allows the tests to run anyway
+    NM_DEVICE_TYPE_WIFI = 2
+
+
+class AvailableWifiNetwork(object):
+    @classmethod
+    def all(cls):
+        result = []
+
+        if not NetworkManager.WirelessEnabled:
```

Comment:
I'd add a check covering case when NetworkManager is not available at all (cf my previous comment :) ).